compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -606,33 +611,39 @@ impl<'ll, 'tcx> DebugInfoMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         variable_kind: VariableKind,
         span: Span,
     ) -> &'ll DIVariable {
-        let loc = self.lookup_debug_loc(span.lo());
-        let file_metadata = file_metadata(self, &loc.file);
-
-        let type_metadata = type_di_node(self, variable_type);
-
-        let (argument_index, dwarf_tag) = match variable_kind {
-            ArgumentVariable(index) => (index as c_uint, DW_TAG_arg_variable),
-            LocalVariable => (0, DW_TAG_auto_variable),
-        };
-        let align = self.align_of(variable_type);
-
-        let name = variable_name.as_str();
-        unsafe {
-            llvm::LLVMRustDIBuilderCreateVariable(
-                DIB(self),
-                dwarf_tag,
-                scope_metadata,
-                name.as_ptr().cast(),
-                name.len(),
-                file_metadata,
-                loc.line,
-                type_metadata,
-                true,
-                DIFlags::FlagZero,
-                argument_index,
-                align.bytes() as u32,
-            )
-        }
+        debug_context(self)
+            .variables
```

Comment:
Why is deduplicating variables useful too?

Reply:
It's required by LLVM when emitting DWARF - there is an assert to ensure that variables are not duplicated.

Reply:
Then, why was it not necessary before this PR?

Reply:
Since we were creating new `DISubprogam` entries per inline site, the variables weren't considered to be duplicates (they were the same named variables in two different functions). Now that we are reusing the same `DISubprogram` LLVM is seeing that we are creating two variables with the same name within the same scope, hence we need to deduplicate the variables as well.